### DISCUSSION TITLE
How to render data above maxzoom ?

### DISCUSSION THREAD

**Discussion by BenOnTrack** [systemed/tilemaker#610] [Q&A] [answered]
Hi,

My entire mbtiles configuration has a `maxzoom` set to 14. 

For instance, let's take the housenumber and the openmaptiles config:

`"housenumber":      { "minzoom": 14, "maxzoom": 14 },`

Library such as maplibre do "overzoom" to be able to render tiles above the maxzoom (up to zoom 24). My current issue is that, in the style.json if you set to display a layer at minzoom > 14, nothing got rendered, as if everything _had to_ be rendered at least at zoom 14 to be able to be "overzoomed".

But what if I _only_ want to display those from zoom 18 ? 

```
 {
      "id": "label_housenumber",
      "type": "symbol",
      "source": "basemap",
      "source-layer": "label",
      "filter": [ "all", ["==", "class", "housenumber"] ],
      "minzoom": 18,
      "layout": {
        "text-font": ["Noto Sans Regular"],
        "text-size": 10,
        "text-field": "{housenumber}"
      },
      "paint": {"text-color": "rgba(119, 102, 85, 0.69)"}
    },
```

**Comment by systemed** [comment]
This should work largely as you've written it but with one alteration.

tilemaker's `config.json` controls which vector tiles the layer is written to. So when you have

    "housenumber":      { "minzoom": 14, "maxzoom": 14 },

that means "write `housenumber` to the z14 tiles" (which is the maximum). All good so far.

Maplibre's `style.json`, which is your second quote here, determines what is rendered at which screen zoom level. So you've set `minzoom` correctly there too.

Where you have a bit of a mistake is here:

      "source-layer": "label",

This means "the source data for this rendering comes from the `label` layer in the vector tiles". But in tilemaker, you wrote it to the `housenumber` layer. So you need to tell the Maplibre style to look in the `housenumber` layer, which you do by setting it to

      "source-layer": "housenumber",

instead.

**Reply by BenOnTrack** [reply]
Oups sorry, I've just copy/paste the openmaptiles config (instead of mine that I've rename "label"), but I confirm I'm actually targeting the right layer, and nothing displays. If I put in the `style.json` minzoom=14 and then "overzoom", everything will be fine. If I put in the `style.json` minzoom>14 and then "overzoom", nothing will show up. 

```
 {
      "id": "label_housenumber",
      "type": "symbol",
      "source": "basemap",
      "source-layer": "label",
      "filter": [ "all", ["==", "class", "housenumber"] ],
      "minzoom": 17,
      "layout": {
        "text-font": ["Noto Sans Regular"],
        "text-size": {
          "stops": [
            [
              17,
              9
            ],
            [
              22,
              11
            ]
          ]
        },
        "text-field": "{housenumber}",
        "text-padding": 3,
        "text-line-height": -0.15,
        "symbol-avoid-edges": false,
        "text-allow-overlap": false,
        "text-ignore-placement": false
      },
      "paint": {"text-color": "rgba(119, 102, 85, 0.69)"}
    },
```

I also forgot to mention that I'm using mbtiles protocol to "access" the tiles, not tilerserver-gl

As of now, the only I found to make it work is by tweaking the text-size

 ```
{
      "id": "label_housenumber",
      "type": "symbol",
      "source": "basemap",
      "source-layer": "label",
      "metadata": {"switch": false},
      "filter": [ "all", ["==", "class", "housenumber"] ],
      "minzoom": 14,
      "layout": {
        "text-font": ["Noto Sans Regular"],
        "text-size": {
          "stops": [
            [
              14,
              0
            ],
            [
              17,
              9
            ],
            [
              22,
              11
            ]
          ]
        },
        "text-field": "{housenumber}",
        "text-padding": 3,
        "text-line-height": -0.15,
        "symbol-avoid-edges": false,
        "text-allow-overlap": false,
        "text-ignore-placement": false
      },
      "paint": {"text-color": "rgba(119, 102, 85, 0.69)"}
    },
```

and so my question: is it actually possible to render data only up to zoom 14 and still display them at zoom above 14 ?

**Reply by BenOnTrack** [reply]
Any feedback on that one? I understand that even if a layer contains data up to zoom 14, it should be able to be rendered at higher zoom level (> 14) by maplibre-gl (for instance, zoom 17)

So it may be linked to the way I access my tiles (working on a PWA with an IndexedDB) - most likely when accessing to tile 17, maplibre has no knowledge that there was data stored at zoom 14 that should now be displayed, as per the config file - so I assume it's being handled by how tool such as tileserver-gl serve the tiles ...

**Reply by BenOnTrack** [reply, marked as answer]
I've found my issue
The style in maplibre should indicate the maxzoom expected

```
"sources": {
    "basemap": {
      "type": "vector",
      "tiles": ["basemap/{z}/{x}/{y}.pbf"],
      "maxzoom":14
    },
```
